SonarQube.TeamBuild.PreProcessor/SonarLintAnalyzerProvider.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        private static bool TryCreateRuleset(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, ILogger logger)
+        {
+            logger.LogDebug(Resources.SLAP_FetchingSonarLintRuleset);
+            string content = TryGetProfileExportForProject(server, projectKey, logger);
+
+            if (content != null)
+            {
+                if (IsValidRuleset(content))
+                {
+                    string ruleSetFilePath = GetRulesetFilePath(settings);
+                    logger.LogDebug(Resources.SLAP_SonarLintRulesetCreated, ruleSetFilePath);
+
+                    File.WriteAllText(ruleSetFilePath, content);
+                    return true;
+                }
+                else
+                {
+                    // TODO: decide on an appropriate error message once we know whether could
+                    // happen in the 1.1 release.
+                    logger.LogError(Resources.SLAP_InvalidRulesetReturned);
+                }
+            }
+            return false;
+        }
+
+        private static string TryGetProfileExportForProject(ISonarQubeServer server, string projectKey, ILogger logger)
+        {
+            string profileContent = null;
+
+            string qualityProfile;
+            if (server.TryGetQualityProfile(projectKey, CSharpLanguage, out qualityProfile))
+            {
+                if (server.TryGetProfileExport(qualityProfile, CSharpLanguage, SonarLintProfileFormatName, out profileContent))
+                {
+                    logger.LogDebug(Resources.SLAP_ProfileExportFound, projectKey);
+                }
+                else
+                {
+                    logger.LogDebug(Resources.SLAP_ProfileExportNotFound, projectKey);
+                }
+            }
+            else
+            {
+                logger.LogDebug(Resources.SLAP_NoProfileForProject, projectKey);
+            }
+            return profileContent;
+        }
+
+        private static bool IsValidRuleset(string rulesetContent)
+        {
+            // TODO: check the file contains a ruleset. There is a platform bug that means the wrong
+            // profile could be exported, depending on which plugins are installed on the server.
+            // Depending on the workaround put into place in the next version of the C# plugin, this
+            // might not occur in practice. If it could still occur then consider making this check more robust.
+            return rulesetContent.Contains("<RuleSet Name=\"Rules for SonarLint");
+        }
+
+        private static string GetRulesetFilePath(TeamBuildSettings settings)
+        {
+            return Path.Combine(settings.SonarConfigDirectory, RoslynCSharpRulesetFileName);
+        }
+
+        private static void FetchBinaries(ISonarQubeServer server, TeamBuildSettings settings, ILogger logger)
+        {
+            // TODO: https://jira.sonarsource.com/browse/SONARCS-556
+        }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Parameter 'logger' of 'SonarLintAnalyzerProvider.FetchBinaries(ISonarQubeServer, TeamBuildSettings, ILogger)' is never used. Remove the parameter or use it in the method body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AReviewUnusedParameters)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Parameter 'server' of 'SonarLintAnalyzerProvider.FetchBinaries(ISonarQubeServer, TeamBuildSettings, ILogger)' is never used. Remove the parameter or use it in the method body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AReviewUnusedParameters)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Parameter 'settings' of 'SonarLintAnalyzerProvider.FetchBinaries(ISonarQubeServer, TeamBuildSettings, ILogger)' is never used. Remove the parameter or use it in the method body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AReviewUnusedParameters)